dango/types/src/bank.rs

```diff
@@ -64,4 +66,14 @@ pub enum QueryMsg {
         start_after: Option<Part>,
         limit: Option<u32>,
     },
+    #[returns(Metadata)]
+    Metadata { denom: Denom },
+}
+
+#[grug::derive(Serde, Borsh)]
+pub struct Metadata {
+    pub name: NonEmpty<String>,
+    pub symbol: NonEmpty<String>,
+    pub description: String,
```

Comment:
Let's design the metadata such that it's compatible with Cosmos SDK: https://github.com/cosmos/cosmos-sdk/blob/main/x/bank/proto/cosmos/bank/v1beta1/bank.proto#L99-L118

and chain-registry: https://github.com/cosmos/chain-registry/blob/master/osmosis/assetlist.json

Reply:
Nevermind, their schema actually isn't good... for example there's no `decimals`; instead they have a `denom_units` which is less straightforward.